File: scripts_python/functions_save.py
import os

#
# The following functions test the read and write rights
#


# Create file in current folder "func_code" (read-only)
def func_save_textfile_func_code(event, context):
    dir_path = os.getcwd()
    print("Try to write into directory " + dir_path)
    f = open(os.path.join(dir_path, 'file.txt'), 'w+')
    f.write('This is the new file.')
    f.close()

    return {
        'status': 200,
        'data': 'OK'
    }


# Create file in subfolder of "func_code" (read-only)
def func_save_textfile_func_code_subfolder(event, context):
    dir_path = os.getcwd()
    print("Try to write into directory " + dir_path)
    os.makedirs(dir_path + "/test")
    f = open(os.path.join(dir_path, 'test/file.txt'), 'w+')
    f.write('This is the new file.')
    f.close()

    return {
        'status': 200,
        'data': 'OK'
    }


# Create file one folder above "func_code" (read and write)
def func_save_textfile_above(event, context):
    path = os.getcwd()
    file_path = os.path.join(path, '..', 'file.txt')
    print("Write following file: " + file_path)
    f = open(file_path, 'w+')
    f.write('This is the new file.')
    f.close()

    return {
        'status': 200,
        'data': 'OK'
    }

File: scripts_ruby/functions_save.rb
#
# The following functions test the read and write rights
#

def create_file(filename)
    File.open(filename, "w") do |f|     
        f.write('This is a new file.')   
    end
end


# Create file in current folder "func_code" (read-only)
def func_save_textfile_func_code(event, context)
    create_file("file.txt")

    return {
        :status => 200,
        :data => 'OK',
    }
end


# Create file in subfolder of "func_code" (read-only)
def func_save_textfile_func_code_subfolder(event, context)
    create_file("test/file.txt")

    return {
        :status => 200,
        :data => 'OK',
    }
end


# Create file one folder above "func_code" (read and write)
def func_save_textfile_above(event, context)
    file = File.join('..', 'file')
    create_file(file)

    return {
        :status => 200,
        :data => 'OK',
    }
end
